gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedSource.java

```diff
@@ -191,23 +209,31 @@ protected void initLogger(SourceState state) {
    */
   public List<String> getcurrentFsSnapshot(State state) {
     List<String> results = new ArrayList<>();
-    String path = state.getProp(ConfigurationKeys.SOURCE_FILEBASED_DATA_DIRECTORY) + "/*"
-        + state.getProp(ConfigurationKeys.SOURCE_ENTITY) + "*";
+    String path = getLsPattern(state);
 
     try {
       log.info("Running ls command with input " + path);
       results = this.fsHelper.ls(path);
       for (int i = 0; i < results.size(); i++) {
-        String filePath = state.getProp(ConfigurationKeys.SOURCE_FILEBASED_DATA_DIRECTORY) + "/" + results.get(i);
+        File parsedFile = new File(new URI(results.get(i)));
```

Comment:
Would this throw an error if the uri points to a relative path but it can't refer to a file without `ConfigurationKeys.SOURCE_FILEBASED_DATA_DIRECTORY`? Should we use `URI#isAbsolute` instead?